### PR TITLE
MGMT-16667: Validate pull secret

### PR DESCRIPTION
**Pull request by eranco74**
Validate pull secret
Pass the pullSecret as part of the cluster config instead of writing the secret to manifest dir

### CODE REVIEW

File: controllers/imageclusterinstall_controller.go [hunk 1]
```diff
@@ -781,6 +761,62 @@ func generateInfraID(base string) string {
 	return fmt.Sprintf("%s-%s", base, utilrand.String(5))
 }
 
+// GetValidPullSecret validates the pull secret reference and format, return the pull secret data
+func (r *ImageClusterInstallReconciler) GetValidPullSecret(ctx context.Context, psRef *corev1.LocalObjectReference, namespace string) (string, error) {
```
Comment (carbonin) [inline controllers/imageclusterinstall_controller.go:765]: No reason for this to be exported (method should be lower case).

File: controllers/imageclusterinstall_controller.go [hunk 2]
```diff
@@ -781,6 +761,62 @@ func generateInfraID(base string) string {
...
+
+	err := r.ValidatePullSecret(string(psData))
+	if err != nil {
+		return "", fmt.Errorf("invalid pull secret data in secret %w", err)
+	}
+	return string(psData), nil
+}
+
+// ImagePullSecret checks if the given string is a valid image pull secret and returns an error if not.
+func (r *ImageClusterInstallReconciler) ValidatePullSecret(pullSecret string) error {
```
Comment (carbonin) [inline controllers/imageclusterinstall_controller.go:787]: Same here, should be `validatePullSecret`

File: controllers/imageclusterinstall_controller.go [hunk 3]
```diff
@@ -781,6 +761,62 @@ func generateInfraID(base string) string {
...
+	errs := []error{}
+
+	for d, a := range s.Auths {
+		_, authPresent := a["auth"]
+		_, credsStorePresent := a["credsStore"]
+		if !authPresent && !credsStorePresent {
+			errs = append(errs, fmt.Errorf("invalid pull secret: %q JSON-object requires either 'auth' or 'credsStore' field", d))
+			continue
+		}
+		data, err := base64.StdEncoding.DecodeString(a["auth"].(string))
```
Comment (carbonin) [inline controllers/imageclusterinstall_controller.go:807]: Is it valid for this not to be present? Might we end up with a panic here for `a["auth"]` being nil?

Comment (eranco74) [reply controllers/imageclusterinstall_controller.go:807]: It's validated few lines above, I'll update the code to use the same var

**Review by carbonin — APPROVED** [review]

**Pull request merged**